{{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_widget.py

```diff
@@ -1,28 +1,125 @@
 """
-This module is an example of a barebones QWidget plugin for napari
+This module contains four napari widgets declared in
+different ways:
 
-It implements the Widget specification.
-see: https://napari.org/stable/plugins/guides.html?#widgets
+- a pure Python function flagged with `autogenerate: true`
+    in the plugin manifest. Type annotations are used by
+    magicgui to generate widgets for each parameter. Best
+    suited for simple processing tasks - usually taking
+    in and/or returning a layer.
+- a `magic_factory` decorated function. The `magic_factory`
+    decorator allows us to customize aspects of the resulting
+    GUI, including the widgets associated with each parameter.
+- a `magicgui.widgets.Container` subclass. This provides lots
+    of flexibility and customization options while still supporting
+    `magicgui` widgets and convenience methods for creating widgets
+    from type annotations.
+- a `QWidget` subclass. This provides maximal flexibility but requires
+    full specification of widget layouts, callbacks, events, etc.
+
+References:
+- Widget specification: https://napari.org/stable/plugins/guides.html?#widgets
+- magicgui docs: https://pyapp-kit.github.io/magicgui/
 
 Replace code below according to your needs.
 """
 from typing import TYPE_CHECKING
 
+import numpy as np
 from magicgui import magic_factory
+from magicgui.widgets import CheckBox, Container, create_widget
 from qtpy.QtWidgets import QHBoxLayout, QPushButton, QWidget
 
 if TYPE_CHECKING:
     import napari
 
 
+# Uses the `autogenerate: true` flag in the plugin manifest
+# to indicate it should be wrapped as a magicgui to autogenerate
+# a widget.
+def example_function_widget(
+    img: "napari.types.ImageData",
+) -> "napari.types.LabelsData":
+    return img > 0.5
```

Comment:
Recommend re-scaling input image data to float before performing the threshold calculation.